src/MapboxStyleParser.ts

```diff
@@ -918,15 +918,12 @@ export class MapboxStyleParser implements StyleParser {
    * @param geoStylerStyle A GeoStylerStyle-Style
    * @return The Promise resolving with a GeoStylerStyle-WriteStyleResult
    */
-  writeStyle(geoStylerStyle: Style): Promise<WriteStyleResult<string>> {
-    return new Promise<WriteStyleResult<string>>(resolve => {
+  writeStyle(geoStylerStyle: Style): Promise<WriteStyleResult<MbStyle>> {
+    return new Promise<WriteStyleResult<MbStyle>>(resolve => {
       const unsupportedProperties = this.checkForUnsupportedProperties(geoStylerStyle);
       try {
         const gsStyle = _cloneDeep(geoStylerStyle);
-        const mapboxStyle: any = this.geoStylerStyleToMapboxObject(gsStyle);
-        const output = this.pretty
-          ? JSON.stringify(mapboxStyle, null, 2)
-          : JSON.stringify(mapboxStyle);
+        const output: any = this.geoStylerStyleToMapboxObject(gsStyle);
```

Comment:
```suggestion
        const output: MbStyle = this.geoStylerStyleToMapboxObject(gsStyle);
```

Reply:
Nice catch. I actually had to omit the `sources`. See 8af3055.